libzfs_core/_libzfs_core.py

```diff
@@ -655,6 +660,276 @@ def lzc_exists(name):
+                record_bytes = os.read(fd, _PIPE_RECORD_SIZE)
+                if not record_bytes:
+                    break
+                (size, _, err, _, _) =  struct.unpack(_PIPE_RECORD_FORMAT, record_bytes)
+                xlate.lzc_list_children_xlate_error(err, name)
+                if size == 0:
+                    break
+                data_bytes = os.read(fd, size)
+                result = {}
+                with nvlist_out(result) as nvp:
+                    ret = _lib.nvlist_unpack(data_bytes, size, nvp, 0)
+                if ret != 0:
+                    raise ZFSError(ret, "Failed to unpack list data")
+                child = result['name']
+                if child != name:
+                    yield child
+        finally:
+            os.close(other_fd)
+            os.close(fd)
+
+    return _iterator()
+
+
+@_uncommitted(lzc_list)
+def lzc_list_snaps(name):
+    '''
+    List the snapshots of the ZFS dataset.
+
+    :param str name: the name of the dataset.
+    :return: an iterator that produces the names of the snapshots.
+    '''
+    (fd, other_fd) = lzc_list(name, {'recurse': 1, 'type': {'snapshot': None}})
+
+    def _iterator():
+        try:
+            while True:
+                record_bytes = os.read(fd, _PIPE_RECORD_SIZE)
+                if not record_bytes:
+                    break
+                (size, _, err, _, _) =  struct.unpack(_PIPE_RECORD_FORMAT, record_bytes)
+                xlate.lzc_list_snaps_xlate_error(err, name)
+                if size == 0:
+                    break
+                data_bytes = os.read(fd, size)
+                result = {}
+                with nvlist_out(result) as nvp:
+                    ret = _lib.nvlist_unpack(data_bytes, size, nvp, 0)
+                if ret != 0:
+                    raise ZFSError(ret, "Failed to unpack list data")
+                snap = result['name']
+                if snap != name:
+                    yield snap
+        finally:
+            os.close(other_fd)
+            os.close(fd)
+
+    return _iterator()
+
+
+def is_supported(func):
+    '''
+    Check whether C *libzfs_core* provides implementation required
+    for the given Python wrapper.
+
+    :param function func: the function to check.
+    :return bool: whether the funciton can be used.
+
+    If `is_supported` returns ``False`` for the function, then
+    calling the function would result in :exc:`NotImplementedError`.
+    '''
+    try:
+        # woulf fail if is_supported is not set yet
+        return func._is_supported
+    except AttributeError:
+        fname = getattr(func, '__name__', None)
+        if fname is None:
+            raise ValueError('argument does not have __name__')
+        if not fname in globals():
+            raise ValueError(fname + ' is not from libzfs_core')
+        if not callable(func):
+            raise ValueError(fname + ' is not a function')
+        func._is_supported = (getattr(_lib, fname, None) != None)
```

Comment:
Usually you want `is not None` - http://stackoverflow.com/questions/11166748/identity-versus-equality-for-none-in-python gives a sort of good explanation about why (ignore the stuff about it "not having a value" - but the rest is valid)